test/governance/StkTruToken.test.ts

```diff
@@ -268,6 +268,40 @@ describe('StkTruToken', () => {
         await expect(stkToken.claimRewards(constants.AddressZero, { gasLimit: 3000000 })).to.be.revertedWith('Token not supported for rewards')
       })
     })
+
+    describe('Claim Restake TRU', () => {
+      beforeEach(async () => {
+        await stkToken.stake(amount, { gasLimit: 3000000 })
+        await timeTravel(provider, DAY)
+        await tfusd.mint(stkToken.address, parseEth(1), { gasLimit: 3000000 })
+      })
+
+      it('emits Claim event', async () => {
+        await expect(stkToken.claimRestakeTRU()).to.emit(stkToken, 'Claim')
+      })
+
+      it('emits Stake event', async () => {
+        await expect(stkToken.claimRestakeTRU()).to.emit(stkToken, 'Stake')
+      })
+
+      it('clears claimable balance', async () => {
+        expectScaledCloseTo(await stkToken.claimable(owner.address, tru.address), parseTRU(1000))
+        await stkToken.claimRestakeTRU()
+        expect(await stkToken.claimable(owner.address, tru.address)).to.equal(0)
+      })
+
+      it('stakes claimable balance', async () => {
+        expect(await stkToken.stakeSupply()).to.equal(amount)
+        await stkToken.claimRestakeTRU()
+        expectScaledCloseTo(await stkToken.stakeSupply(), amount.add(parseTRU(1000)))
```

Comment:
Don't actually know if it is necessary, but adding an extra test never did any harm. We could maybe do the same thing with timeTravel and a little increased value? (proportionally to the distributor's distribution of course)

Anyway, the code looks good 🥳

Reply:
Thanks!

With this test I'm just making sure the basic functionality of this function gets covered, so a refactor later on doesn't break this without warning us. timeTravel might also be nice, but as long as it's tested more carefully elsewhere, I don't think there's much need to repeat it here.

I'm not too attached to this opinion though, so if you'd like I can add a test case.